### ISSUE TITLE
Shared With Me folders without a ParentReference behave oddly when being navigated

### ISSUE THREAD

**Issue by Esvandiary** [KoenZomers/KeePassOneDriveSync#55]
Conditions: a shared folder is owned by a OneDrive for Business group (effectively a SharePoint site), rather than another user.

If I have a site SomeSite with a top-level folder ABC, and then share ABC with myself, as expected the ABC folder appears in the OneDrive Shared With Me section (both in the OneDrive UI and the plugin).
However, if I double-click on the ABC folder within the plugin's Shared With Me file selection dialog, I instead get a view of the SomeSite root (i.e. I can still see an ABC folder, along with any other folders at that level). Navigation from this point on is then normal, and I can select a file and use the plugin as expected.

This issue is only visible once #56 is applied (since otherwise a NullReferenceException is thrown before reaching this point)

**Comment by Esvandiary** [comment]
After the fix in #56, although there are no longer any errors, the browsing behaviour is still slightly odd. (Hence, I made a separate issue to hopefully track this)

If I have a site SomeSite with a top-level folder ABC, and then share ABC with myself, as I'd expect the ABC folder appears in the OneDrive Shared With Me section (both in the OneDrive UI and the plugin).
However, if I double-click on the ABC folder within the plugin's Shared With Me file selection dialog, I instead get a view of the SomeSite root (i.e. I can still see an ABC folder, along with any other folders at that level). Navigation from this point on is then normal, and I can select a file and use the plugin as expected.

**Comment by KoenZomers** [comment]
You did a great job already detailing the steps, but could you add a little bit more detail even so I can try to reproduce it and have a look at the issue as well? In your steps can you at least answer these questions:

1. Does it happen with SharePoint on-premises and/or SharePoint Online?
2. Do I need to create a sitecollection with a specific template? I.e. TeamSite? Or is it independent on the chosen template?
3. Are you using the Graph API connector or the OneDrive for Business connector in KeePass to set up the sync?
4. The top level folder ABC, did you create this in the default document library inside the site collection or did you use i.e. SharePoint Designer to create a folder in the root of the site collection?

**Comment by Esvandiary** [comment]
1. My test was done via an Office 365 for Business account, and thus was SharePoint Online; I'm afraid I don't have access to an on-premises SharePoint setup to test with.
2. I created the site via the Office 365 OneDrive's "group" option, which I believe creates a SharePoint Team Site under the hood. I haven't tested with other setups, but I could try if you have a suggested config. (I'm quite new to Office 365/SharePoint administration, so you might have to bear with me...)
3. I was using the Graph API connector. During the initial testing of the #56 crash I got the same results using both the Graph API and OneDrive for Business connectors, but I don't recall if I retested using the OneDrive for Business connector after fixing the crash; I'll give that another try when I have time.
4. I created a folder inside the default Documents library via the OneDrive web interface.
